mesonbuild/environment.py

```diff
@@ -696,6 +697,9 @@ def _guess_nix_linker(compiler: typing.List[str], for_machine: MachineChoice,
             `-Xlinker=--version`) you must pass as a list.
         :extra_args: Any addtional arguments rquired (such as a source file)
         """
+        if CudaCompiler.cuda_id() in compiler:
+            return CudaLinker(compiler, for_machine, 'nvlink', prefix, version=CudaLinker.parse_version())
+
```

Comment:
I don't think this should go in guess_nix_linker, since nvcc will always be paired with nvlink, correct? If that's the case I'd rather do what we do for other 1:1 compiler/linker pairs (like pgi, intel, and ccrx) and just always create the the two together.

Reply:
I will look how meson handles those compilers and mimic it for cuda.
Creating them together would eliminate the need for this check and for the static member from the previous comment.

Reply:
Done as recomended